lib/util.js

```diff
@@ -1002,9 +1002,12 @@ function error(...args) {
 }
 
 function _errnoException(err, syscall, original) {
-  if (typeof err !== 'number' || err >= 0 || !Number.isSafeInteger(err)) {
-    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'err',
-                               'negative number');
+  if (typeof err !== 'number') {
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'err', 'number', err);
+  }
+  if (err >= 0 || !Number.isSafeInteger(err)) {
+    throw new errors.RangeError('ERR_VALUE_OUT_OF_RANGE', 'err',
+                                'a negative number', err);
```

Comment:
Since there is an integer check and the type name is already in prose style, maybe change this to `'a negative integer'` for better clarity?